### PR TITLE
add mutation for register

### PR DESCRIPTION
**Pull request by u-uysal**

### CODE REVIEW

File: src/Models/User/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { model, Schema } from 'mongoose';
+
+const userSchema = new Schema({
+  username: String,
+  password: String,
+  email: String,
+  createdAt: String,
```
Comment (rizikolik) [inline src/Models/User/index.js:7]: Not Date ?

File: src/Utils/validator.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { UserInputError } from 'apollo-server';
+
+const validateRegisterData = (username, email, password, confirmPassword) => {
+  if (username.trim() === '') {
+    throw new UserInputError('Username must not be empty');
```
Comment (rizikolik) [inline src/Utils/validator.js:5]: I prefer, `should not` instead of `must not` or 
`User name cant be empty`

File: src/Utils/validator.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (email.trim() === '') {
+    throw new UserInputError('Email must not be empty');
+  } else {
+    const regEx = /^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*@([0-9a-zA-Z][-\w]*[0-9a-zA-Z]\.)+[a-zA-Z]{2,9})$/;
+    if (!email.match(regEx)) {
+      throw new UserInputError('Email must be a valid email address');
+    }
+  }
+  if (password === '') {
+    throw new UserInputError('Password must not empty');
```
Comment (rizikolik) [inline src/Utils/validator.js:16]: Same

File: src/graphql/User/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import bcrypt from 'bcryptjs';
+import gql from 'graphql-tag';
+import jwt from 'jsonwebtoken';
+import logger from '../../../logger';
+import User from '../../Models/User';
+import validateRegisterData from '../../Utils/validator';
+
+export const UserTypeDef = gql`
+  # extend type Query {
+  #   book: String
```
Comment (rizikolik) [inline src/graphql/User/index.js:11]: book ?

File: src/Models/User/index.js (hunk 1 above)
Comment (u-uysal) [reply src/Models/User/index.js:7]: I would like to keep it as a string. Thats why I used `new Date().toISOString()`

File: src/graphql/User/index.js (hunk 4 above)
Comment (u-uysal) [reply src/graphql/User/index.js:11]: I will keep it as example to not forget :D

**Pull request merged**